File: Python/euklides.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


def euklides(a, b):
    while a != b:
        if a > b:
            a = a - b
        else:
            b = b - a
    return a

    def euklides2(a, b):
        while a > 0:
            a = a % b
            b = b - a
    return b


def main(args):
    a = int(input("Podaj liczbę: "))
    b = int(input("Podaj liczbę: "))
    assert euklides(1989, 867) == 51
    assert euklides(12, 6) == 6
    assert euklides2(1989, 867) == 51
    assert euklides2(12, 6) == 6
    print("Nwd({:d}, {:d}) = {:d}".format(a, b, euklides2(a, b)))

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/euklides.cpp
#include <iostream>

using namespace std;

int euklides(int a, int b) {
    while(a != b)
    {
        if (a > b)
            a = a - b;
        else
            b = b - a;
    }
    
    }

int main(int argc, char **argv)
{
    
    cout << "Podaj liczbę: ";
    cin >> a;
    cout << "Podaj liczbę: ";
    cin >> b;
    
    cout << "Największy wspólny dzielnik  =  " << a << endl;
   

    return 0;
}

